test/dlc_tests/conftest.py

```diff
@@ -222,10 +236,210 @@ def pytest_runtest_setup(item):
+                str_keywords,
+            )
+            model_scope = _infer_field_value(
+                "N/A", ("mnist", "densenet", "squeezenet", "half_plus_two", "half_plus_three"), str_keywords
+            )
+            num_instances = _infer_field_value(
+                1, ("_multinode_", "_multi-node_", "_multi_node_"), str_fspath, str_keywords
+            )
+            processor_scope = _infer_field_value("all", ("cpu", "gpu", "eia"), str_keywords)
+            if processor_scope == "gpu":
+                processor_scope, failure_conditions = _handle_single_gpu_instances(
+                    function_key, str_keywords, failure_conditions
+                )
+
+            # Create a new test coverage item if we have not seen the function before. This is a necessary step,
+            # as parametrization can make it appear as if the same test function is a unique test function
+            test_cov[function_key] = {
+                "Category": category,
+                "Name": function_name,
+                "Scope": framework_scope,
+                "Job_Type": job_type_scope,
+                "Num_Instances": get_marker_arg_value(item, "multinode", num_instances),
+                "Processor": processor_scope,
+                "Integration": get_marker_arg_value(item, "integration", integration_scope),
+                "Model": get_marker_arg_value(item, "model", model_scope),
+                "GitHub_Link": github_link,
+            }
+        write_test_coverage_file(test_cov, test_coverage_file)
+
+        if failure_conditions:
+            message, total_issues = _assemble_report_failure_message(failure_conditions)
+            if total_issues == 0:
+                LOGGER.warning(f"Found failure message, but no issues. Message:\n{message}")
+            else:
+                raise TestReportGenerationFailure(message)
+
+
+def _handle_single_gpu_instances(function_key, function_keywords, failures, processor="gpu"):
+    """
+    Generally, we do not want tests running on single gpu instance types. However, there are exceptions to this rule.
+    This function is used to determine whether we need to raise an error with report generation or not, based on
+    whether we are using single gpu instances or not in a given test function.
+
+    :param function_key: local/path/to/function::function_name
+    :param function_keywords: string of keywords associated with the test function
+    :param failures: running dictionary of failures associated with the github link
+    :param processor: whether the test is for cpu, gpu or both
+    :return: processor if not single gpu instance, else "single_gpu", and a dict with updated failure messages
+    """
+
+    # Define conditions where we allow a test function to run with a single gpu instance
+    whitelist_single_gpu = False
+    allowed_single_gpu = ("telemetry", "test_framework_version_gpu")
```

Comment:
Should this list be placed in a more accessible global location? IMO that'll help making the change quickly for someone who creates a PR.

Reply:
sure, I can create a global for this in conftest